src/main/kotlin/io/github/inductiveautomation/kindling/gatewaynetwork/GatewayNetworkViewer.kt

```diff
@@ -0,0 +1,161 @@
+package io.github.inductiveautomation.kindling.gatewaynetwork
+
+import com.formdev.flatlaf.extras.FlatSVGIcon
+import io.github.inductiveautomation.kindling.core.Kindling.Preferences.General.DefaultEncoding
+import io.github.inductiveautomation.kindling.core.Tool
+import io.github.inductiveautomation.kindling.core.ToolPanel
+import io.github.inductiveautomation.kindling.utils.FileFilter
+import io.github.inductiveautomation.kindling.utils.FlatScrollPane
+import org.json.JSONException
+import org.json.JSONObject
+import java.awt.Desktop
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
+import java.io.File
+import java.net.URI
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.UUID
+import javax.swing.JButton
+import javax.swing.JOptionPane
+import javax.swing.JTextArea
+import kotlin.io.path.name
+import kotlin.io.path.useLines
+
+/**
+ * Opens the raw file that contains a gateway network diagram as JSON text. After opening, the user can click a
+ * button to open a browser that will show the actual diagram. The diagram is served from html and javascript files
+ * copied out to a temp folder on the local file system. The source of the static files is the
+ * cytoscape-server IA internal repository, in the 'cytoscape-server.zip' file. Within the zip file,
+ * navigate to 'server/static' to view the actual files.
+ *
+ * To get a JSON diagram in Ignition 8.1:
+ * Set the 'gateway.routes.status.GanRoutes' logger set to DEBUG in an Ignition gateway to generate diagram JSON while
+ * viewing the gateway network live diagram page.
+ */
+class GatewayNetworkViewer(path: Path) : ToolPanel() {
+    override val icon = GatewayNetworkTool.icon
+
+    private val browserBtn = JButton("View diagram in browser")
+
+    init {
+        name = path.name
+        toolTipText = path.toString()
+        add(browserBtn)
+
+        val textArea = JTextArea()
+        add(FlatScrollPane(textArea), "newline, push, grow, span")
+
+        // Read the provided file and populate the text area with it
+        path.useLines(DefaultEncoding.currentValue) { lines -> parseLines(lines, textArea) }
+
+        browserBtn.addMouseListener(
+            object : MouseAdapter() {
+                override fun mousePressed(e: MouseEvent?) {
+                    // We need to load the contents from the text area and place them in a file
+                    // with the other static files
+                    val theText = textArea.text
+
+                    // Verify that the provided text is valid JSON, as the user is free to copy and paste in text.
+                    try {
+                        JSONObject(theText)
```

Comment:
Let's use `kotlinx.serialization.json.JsonObject` here instead of the `org.JSON` import.
It would also be possible to verify it's a complete, accurate GAN export with some simple `@Serializable` data classes to represent the data model; see the thread model for an example of that